logic/dungeon7.py

```diff
@@ -5,59 +5,188 @@
+        pegs_after_a_stairs.connect(east_ledge, None, one_way=True) # On Pegs Around Chest --> East Beamos Ledge
+
+        # floor 2
+        after_a_stairs.connect(ne_pillar, None) # Ball Room <--> Northeast Pillar Area #TODO: delete in favor of casual logic commented out below
+        ne_pillar.connect(ne_pillar_chest5, POWER_BRACELET) # Northeast Pillar Area <--> Horse Head, Bubble Chest
+        after_b_stairs.connect(ne_pillar, r.hit_switch) # Horseheads Staircase <--> Northeast Pillar Area
+        ne_pillar.connect(se_pillar, FEATHER) # Northeast Pillar Area <--> Southeast Pillar Area
+        after_c_stairs.connect(after_c_stairs_chest2, r.enemy_requirements["THREE_OF_A_KIND"]) # North Three-of-a-Kind Puzzle <-->  #TODO: When ROM patched to reset ball on S&Q in ROM, then add bracelet method
+        after_c_stairs.connect(spike_corridor, FEATHER) # [logic prep for staircase rando]
+        spike_corridor.connect(se_pillar, FEATHER, one_way=True) # Between Pillar Pushbocks --> Southeast Pillar Area
+        nw_pillar.connect(spike_corridor, FEATHER, one_way=True) # Northwest Pillar Area --> Between Pillar Pushbocks
+        after_d_stairs.connect(tile_room, None, one_way=True) # Hinox Area --> Floating Tile Fight
+        tile_room.connect(nw_pillar, None) # Floating Tile Fight <--> Northwest Pillar Area
+        after_d_stairs.connect(after_d_stairs_drop2, r.miniboss_requirements["HINOX"]) # Hinox Area <--> Hinox Key
+        after_d_stairs.connect(keylock_ledge, FOUND(KEY7, 3)) # Hinox Area <--> Key Locked Ledge
+        after_d_stairs.connect(pegs_after_a_stairs, OR(r.hit_switch)) # Hinox Area <--> On Pegs Around Chest #TODO: REPLACE with below
+        #TODO: after_a_stairs.connect(pegs_after_a_stairs, BOOMERANG) # Ball Room <--> On Pegs Around Chest # NEW VERSION
+        #TODO: before_b_stairs.connect(pegs_after_a_stairs, OR(BOOMERANG, BOW, BOMB, MAGIC_ROD, COUNT(SWORD,2))) # Hinox Area <--> On Pegs Around Chest # NEW VERSION
+        keylock_ledge.connect(pegs_after_a_stairs, None, one_way=True) # Key Locked Ledge --> On Pegs Around Chest
+        keylock_ledge.connect(se_pillar, None, one_way=True) # Key Locked Ledge --> Southeast Pillar Area
+        keylock_ledge.connect(after_d_stairs, None, one_way=True) # Key Locked Ledge --> Hinox Area
+        pegs_after_a_stairs.connect(after_d_stairs, None, one_way=True) # On Pegs Around Chest --> Hinox Area
+        pegs_after_a_stairs.connect(pegs_after_a_stairs_chest7, None) # On Pegs Around Chest --> Mirror Shield Chest
+        nw_pillar.connect(bomb_corridor, BOMB) # Northwest Pillar Area <--> Bombable Wall Corridor
+        bomb_corridor.connect(bombwall_owl, BOMB) # Bombable Wall Corridor <--> By Pit Owl
+        #TODO: bomb_corridor.connect(after_d_stairs, None, one_way=True) # Bombable Wall Corridor --> Hinox Area # push blocks #[logic prep for staircase rando]
+        bombwall_owl.connect(sw_pillar, HOOKSHOT) # By Pit Owl <--> Southwest Pillar Area

```

Comment:
In order to hookshot across the 4 tile pit gap from bombwall_owl to sw_pillar you need the chest to have spawned, so this would require r.enemy_requirements["THREE_OF_A_KIND"] already.